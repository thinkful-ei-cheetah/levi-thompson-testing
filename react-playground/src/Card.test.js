import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Card from './Card.js';

describe('<Card />', () => {
    //------- Smoke Test -------//
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    //------- Snapshot Test -------//
    

    it('renders looks as expected', () => {
        const id = '1234';
        const title = 'Test Title';
        const content = 'Test Content'; 
        const tree = renderer.create(<Card key={id} title={title} content={content}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
})