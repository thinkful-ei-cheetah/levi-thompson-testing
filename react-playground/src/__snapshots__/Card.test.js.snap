// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders looks as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    Test Title
  </h3>
  <p>
    Test Content
  </p>
</div>
`;
